File: TransCoder_tokenized_test_set_functions/python/CONVERT_SUBSTRINGS_LENGTH_K_BASE_B_DECIMAL.py
def substringConversions ( s, k, b ) :
    l = len ( s );
    
    for i in range ( l ) :
        if ( ( i + k ) < l + 1 ) :
            sub = s [ i : i + k ];
            
            sum, counter = 0, 0;
            
            for i in range ( len ( sub ) - 1, - 1, - 1 ) :
                sum = sum + ( ( ord ( sub [ i ] ) - ord ( '0' ) ) * pow ( b, counter ) );
                
                counter += 1;
                
            print ( sum, end = " " );
            
    

File: TransCoder_tokenized_test_set_functions/cpp/CONVERT_SUBSTRINGS_LENGTH_K_BASE_B_DECIMAL.cpp
int substringConversions ( string str, int k, int b ) {
  for ( int i = 0;
  i + k <= str . size ( );
  i ++ ) {
    string sub = str . substr ( i, k );
    int sum = 0, counter = 0;
    for ( int i = sub . size ( ) - 1;
    i >= 0;
    i -- ) {
      sum = sum + ( ( sub . at ( i ) - '0' ) * pow ( b, counter ) );
      counter ++;
    }
    cout << sum << " ";
  }
}